### PR TITLE
Add an example C++ program which fetches URLs from the command line

### PR DESCRIPTION
**Pull request by RyanTheOptimist**
Risk Level: None
Testing: None
Docs Changes: Where would the best place to add them be?

### CODE REVIEW

**Review by alyssawilk — COMMENTED** [review]
couple of questions
first, what's the motivation behind checking this in?
second, do we do HTTP/3 by default?  I thought there was work to turn it up and might be worth doing that
third because my brain isn't working are the requests done in parallel or serial? let's comment that and maybe usage at the top of main.cc
/wait

**Comment by RyanTheOptimist** [comment]
> couple of questions first, what's the motivation behind checking this in? 

Ah, good question. Primarily, because we don't have an example app that uses the C++ APIs, only the higher level language APIs.

> second, do we do HTTP/3 by default? I thought there was work to turn it up and might be worth doing that 

As far as I know, no, we don't do it by default. But it should be easy enough to do that. I'll take a stab.

> third because my brain isn't working are the requests done in parallel or serial? let's comment that and maybe usage at the top of main.cc /wait

Done!

**Review by alyssawilk — DISMISSED** [review]
LGTM and I'm fine with H3 as a follow-up

File: mobile/examples/cc/fetch_client/fetch_client_main.cc [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#include "examples/cc/fetch_client/fetch_client.h"
+
+extern const char build_scm_revision[];
+extern const char build_scm_status[];
+
+const char build_scm_revision[] = "0";
+const char build_scm_status[] = "test";
+
+// Fetches each URL specified on the command line in series,
+// and prints the countents to standard out.
```
[the remark this answers is not in the document]
Comment (RyanTheOptimist) [reply mobile/examples/cc/fetch_client/fetch_client_main.cc]: Done!

**Comment by RyanTheOptimist** [comment]
@jpsim friendly ping

**Review by jpsim — APPROVED** [review]
Nice example app. It'd be good to build this on CI, otherwise this could stop building at some point.

For the iOS & Android examples, we actually run them and check the output to make sure it worked end-to-end. You might want to consider doing the same for this.

As for docs, maybe this should be considered the C++ "Hello World" example and docs added here? https://envoymobile.io/docs/envoy-mobile/latest/start/examples/hello_world.html

**Comment by RyanTheOptimist** [comment]
> Nice example app. It'd be good to build this on CI, otherwise this could stop building at some point.
> 
> For the iOS & Android examples, we actually run them and check the output to make sure it worked end-to-end. You might want to consider doing the same for this.
> 
> As for docs, maybe this should be considered the C++ "Hello World" example and docs added here? https://envoymobile.io/docs/envoy-mobile/latest/start/examples/hello_world.html

Thanks for the review. Great suggestions! I'll address both in a followup.

**Pull request merged**